bin/test_exercise.sh

```diff
@@ -1,112 +1,81 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
-# Test an exercise
+cd "$(git rev-parse --show-toplevel)"
 
-# which exercise are we testing right now?
-# if we were passed an argument, that should be the
-# exercise directory. Otherwise, assume we're in
-# it currently.
-if [ $# -ge 1 ]; then
-    exercise=$1
-    # if this script is called with arguments, it will pass through
-    # any beyond the first to cargo. Note that we can only get a
-    # free default argument if no arguments at all were passed,
-    # so if you are in the exercise directory and want to pass any
-    # arguments to cargo, you need to include the local path first.
-    # I.e. to test in release mode:
-    # $ test_exercise.sh . --release
-    shift 1
-else
-    exercise='.'
+if [ $# -eq 0 ]; then
+    echo "Usage: <exercise-slug> [cargo args]"
+    exit 1
+fi
+
+slug="$1"
+cargo_args="$2"
+
+# determine the exercise path from the slug
+for p in exercises/{practice,concept}/* ; do
+    if [[ "$p" =~ $slug ]]; then
+        exercise_path=$p
+        break
+    fi
+done
+if [ -z "$exercise_path" ]; then
+    echo "Could not find exercise path for $slug"
+    exit 1
 fi
 
 # what cargo command will we use?
 if [ -n "$BENCHMARK" ]; then
-    command="bench"
+    command="+nightly bench"
```

Comment:
`+nightly` is not necessary in CI, where only the nightly toolchain is installed for the benchmark job. But this allows running benchmarks locally as well, where multiple toolchains are probably installed.